packages/medusa/src/repositories/product.ts

```diff
@@ -353,6 +353,18 @@ export class ProductRepository extends Repository<Product> {
     options: FindWithoutRelationsOptions = { where: {} },
     relations: string[] = []
   ): Promise<[Product[], number]> {
+    const tags = options.where.tags
+    delete options.where.tags
+
+    const price_lists = options.where.price_list_id
+    delete options.where.price_list_id
+
+    const sales_channels = options.where.sales_channel_id
+    delete options.where.sales_channel_id
+
+    const discount_condition_id = options.where.discount_condition_id
+    delete options.where.discount_condition_id
```

Comment:
suggestion(non-blocking): would it make sense to extract this to a separate method in order to not duplicate the extraction of properties + the `if(tags)..` etc further down?

Reply:
I definitely would like to do that, but I have tried to do it in a couple of different ways, and it results in the conditions not being appropriately applied. 

I have this test case:

**Price List A**: { id: "price-list-test" }

**Product A**:  { title: "MedusaHeadphones", variants: [{ id: "variant_one", prices: [ { price_list_id: "price-list-test }]}]} << Has relationship to Price List A.

**Product B**: { title: "OtherHeadphones, variants: [{ id: "variant_two" }] } << No relationship to Price List B

With the current implementation using the endpoint `/admin/price-lists/:id/products?q=Headphones` correctly returns _only_ Product A. 

If I then move applying the `.join`'s and `.andWhere`'s into a separate function so that it can be reused in both `getFreeTextSearchResultsAndCount` and `queryProducts` then the above test cases starts to fail.

Reply:
alright, lets keep it as is for now until we find a better way 😄